nodeJS/express_and_mongoose/express_102_crud_and_mvc.md

```diff
@@ -19,58 +19,60 @@ This section contains a general overview of topics that you will learn in this l
 
 We'll need two views, 1) the index, and 2) the display-photo view which will just display a single photo.
 
 Our controller then would be called by Express whenever we get an `app.get()` request. It would then use the details of the request to determine which view is shown, and which image is displayed depending on whether the user is requesting the index or a specific photo's page.
 
-If this is a little confusing at this point, don't worry about it too much.  You will be creating models, views, and controllers in the tutorial and it will all become much clearer once you see them in use.
+If this is a little confusing at this point, don't worry about it too much. You will be creating models, views, and controllers in the tutorial and it will all become much clearer once you see them in use.
 
 ### Which database should you choose?
 
-One final note before diving back into the tutorial.  Express does not care about which database you use.  The lesson lists a few options but ultimately uses MongoDB. In this case, the actual DB you use matters little.  If you later decide that you would rather use SQL or something else, you should be able to pick it up fairly easily by reading the documentation. At this point, Mongo is probably the most popular choice to use with Express so we recommend just sticking with that for now.
+One final note before diving back into the tutorial. Express does not care about which database you use. The lesson lists a few options but ultimately uses MongoDB. In this case, the actual DB you use matters little. If you later decide that you would rather use SQL or something else, you should be able to pick it up fairly easily by reading the documentation. At this point, Mongo is probably the most popular choice to use with Express so we recommend just sticking with that for now.
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
+1. If you're planning to publish your code on GitHub you will need to hide the MongoDB connection URL beacuse it contains your username and password. One way to do this is to push your code to a private repository. However, if you want to show-off your code, you can assign your connection url to an environment variable and access it with [the Dotenv package](https://github.com/motdotla/dotenv). This[Medium article](https://javascript.plainenglish.io/how-to-use-dotenv-and-why-you-need-it-in-all-your-projects-a7319e16124a) does a great job of explaining what environment variables are and how to use them with Express. Read through the article and then when you reach the section that has you connect to MongoDB, follow the instructions in this article to hide your connection string and access it from the `.env` file.
+
 1. Complete part 3 of [the MDN Express tutorial](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/mongoose).
 
 </div>
 
 ### Knowledge checks
```

Comment:
```suggestion
### Knowledge check
```
This is what is causing the error. The custom rule error is not very clear.

Reply:
I missed that typo and some of the rules are a little confusing. Thank you so much for your help!